lib/thread.c

```diff
@@ -1152,6 +1177,9 @@ thread_fetch (struct thread_master *m, struct thread *fetch)
           thread_process_fd (&m->write, &writefd, &m->writefd);
         }
 
+      /* Verify connection status of each socket that left in read list. */
+      thread_connect_check(&m->read, &m->readfd);
```

Comment:
Your thread_connect_check function looks very similar to thread_process_fd() function. The main difference there is you read/reset socket error on all sockets in a reading list. Probably it's better to extend thread_process_fd() function with your logic? What do you think?

Also, can you please explain, why do you need to read/reset error here? What is the purpose here?